Support yofi.toml as config filename

jdek opened the pull request:
Allows editors to more easily syntax highlight. Didn't open an issue beforehand since this feature is so minor.

jdek committed:
config: support yofi.toml filename

Allows text-editors to recognise syntax without extra input from the
user.

jdek committed:
flake: add clippy, rust-analyzer, and rustfmt

l4l commented:
Thanks for your contribution, however I would really prefer avoiding that patch as the config path becomes more obscure, e.g having both `yofi.config` & `yofi.toml` leads to a troubles in figuring out which one is used. I agree the toml extension would be clearer choice but having two implicit defaults or changing one completely lacks of good motivation so far.

Besides, I suppose 99% of editors allows to specify language/syntax for arbitrary file extensions, doesn't it?

jdek committed:
config: deprecate yofi.config

Prints a warning if yofi.config is loaded, please migrate your
configurations by renaming yofi.config -> yofi.toml.
